docs/extraction.md

```diff
@@ -9,6 +9,64 @@ Information extraction is a central capability of the GenAIIDP solution, transfo
 
 Configure extraction behavior through several components:
 
+### Agentic Extraction (Recommended for Production)
+
+The extraction service supports two modes: **traditional** and **agentic**. Agentic extraction provides superior accuracy and consistency, especially for complex documents with nested structures or strict schema requirements.
+
+#### When to Enable Agentic Extraction
+
+Enable agentic extraction when you need:
```

Comment:
Can we say something about this enabling future extensions involving MCP servers to integrate validation or enrichment into the extraction process... (eg validate names, addresses, etc from external services)